yapapi/services.py

```diff
@@ -338,6 +360,26 @@ def _get_handler(instance: ServiceInstance):
         if handler:
             return handler()
 
+    @staticmethod
+    def _change_state(
```

Comment:
If we intend to keep this mode of operation (i.e. not specifying the state to go to but advancing along some predefined order) I'd argue the name `_advance_state` would better describe what's happening here.
Somewhat related: what's the reason behind this "advancing" approach?

Reply:
@zakaprov there's a historical, or rather "evolutional", explanation of why I "adopted" this approach: Parts of this method's code were originally located in few places in the `_run_instance()` method. Then I gathered them into an internal function in `_run_instance()`, called  `_change_state()`. Then, following a suggestion in the code review, I extracted some code from that method to a top-level method.

Now, here's an _a posteriori_ justification of why this approach is good: The method takes as arguments the current state of an instance and information on the events that lead to a state transition: it's either `stop`, denoting a transition due to `stop` signal, or `exc_info`, denoting a transition due to an error, or none, denoting a transition caused by the handler function for the current state terminating normally. Together, the current state and the information on the event (type) are sufficient to determine the target state -- so specifying this state in addition would be simply redundant.

Reply:
@zakaprov I've changed the parameters for `Cluster._change_state()` in https://github.com/golemfactory/yapapi/pull/498/commits/a6789ed0784bbda650ba5cb5c7cdac3d73835c06, perhaps this addresses your concerns?

Reply:
@azawlocki Thanks for the detailed explanation! And yes, with the changes from a6789ed I think it's clearer what this function does (the signature is now simpler and the docstring more informative).